FTXUI fails to compile

**Describe the bug**
The third_party dependency FTXUI fails to compile.

**To Reproduce**
Steps to reproduce the behavior:
1. git clone https://github.com/eclipse-ecal/ecal
2. git submodule init && git submodule update
3. cd ecal && mkdir _build && cd _build
4. cmake .. -DCMAKE_BUILD_TYPE=Release -DECAL_THIRDPARTY_BUILD_PROTOBUF=OFF -DECAL_THIRDPARTY_BUILD_CURL=OFF -DECAL_THIRDPARTY_BUILD_HDF5=OFF -DECAL_THIRDPARTY_BUILD_QWT=OFF
5. make VERBOSE=1
7. Build fails with:
```
[  1%] Building CXX object thirdparty/ftxui/CMakeFiles/dom.dir/src/ftxui/dom/automerge.cpp.o
cd /tmp/ecal/_build/thirdparty/ftxui && /usr/bin/c++  -I/tmp/ecal/thirdparty/ftxui/include -I/tmp/ecal/thirdparty/ftxui/src -O3 -DNDEBUG   -Wall -Wextra -std=c++14 -Wall -Wextra -pedantic -Werror -Wmissing-declarations -Wdeprecated -Wshadow -MD -MT thirdparty/ftxui/CMakeFiles/dom.dir/src/ftxui/dom/automerge.cpp.o -MF CMakeFiles/dom.dir/src/ftxui/dom/automerge.cpp.o.d -o CMakeFiles/dom.dir/src/ftxui/dom/automerge.cpp.o -c /tmp/ecal/thirdparty/ftxui/src/ftxui/dom/automerge.cpp
In file included from /tmp/ecal/thirdparty/ftxui/include/ftxui/dom/elements.hpp:162,
                 from /tmp/ecal/thirdparty/ftxui/src/ftxui/dom/automerge.cpp:4:
/tmp/ecal/thirdparty/ftxui/include/ftxui/dom/take_any_args.hpp: In function ‘ftxui::Elements ftxui::unpack(Args ...)’:
/tmp/ecal/thirdparty/ftxui/include/ftxui/dom/take_any_args.hpp:27:33: error: fold-expressions only available with ‘-std=c++17’ or ‘-std=gnu++17’ [-Werror]
   27 |   (Merge(vec, std::move(args)), ...);
      |                                 ^~~
```

**Expected behavior**
Should build.

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Version 22.04
 - gcc 11.2.0

**Additional context**
It seems that the "-std=c++14" compiler flag is populated all the way through to ftxui from the ecal main CMakeLists.txt.
Removing the two lines make it compile successfully.
```
  set(CMAKE_CXX_STANDARD 14)
  add_definitions(-Wall -Wextra -std=c++14)
```

I am wondering why it doesn't fail in the github pipelines.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,19 +45,11 @@ if(MSVC)
 endif()
 
 # --------------------------------------------------------
-# set mingw specific options
-# --------------------------------------------------------
-if(MINGW)
-  add_definitions(-std=c++14 -DWINVER=0x0601)
-endif()
-
-# --------------------------------------------------------
 # set gcc specific options
 # --------------------------------------------------------
 if(UNIX)
   message(STATUS "GCC detected - Adding flags")
-  set(CMAKE_CXX_STANDARD 14)
-  add_definitions(-Wall -Wextra -std=c++14)
+  add_definitions(-Wall -Wextra)
 
   set(ATOMIC_TEST_CODE "
       #include <atomic>

--- a/samples/CMakeLists.txt
+++ b/samples/CMakeLists.txt
@@ -18,7 +18,6 @@
 cmake_minimum_required(VERSION 3.10)
 
 project(eCAL_samples VERSION "${ECAL_VERSION_STRING}")
-set (CMAKE_CXX_STANDARD 14)
 
 if(HAS_HDF5)
 set(SAMPLE_PERSON_MEASUREMENT_PATH ${CMAKE_CURRENT_SOURCE_DIR}/data/person)

